src/Razor/src/Microsoft.AspNetCore.Razor.Tools/GenerateCommand.cs

```diff
@@ -214,24 +214,25 @@ protected override bool ValidateArguments()
             foreach (var result in results)
             {
                 var errorCount = result.CSharpDocument.Diagnostics.Count;
-                if (errorCount > 0)
+                for (var i = 0; i < errorCount; i++)
                 {
-                    success = false;
+                    var error = result.CSharpDocument.Diagnostics[i];
+                    if (error.Severity == RazorDiagnosticSeverity.Error)
+                    {
+                        success = false;
+                    }
 
-                    for (var i = 0; i < errorCount; i++)
+                    Error.WriteLine(error.ToString());
+
+                    // Only show the first 100 errors to prevent massive string allocations.
+                    if (i == 99)
                     {
-                        var error = result.CSharpDocument.Diagnostics[i];
-                        Error.WriteLine(error.ToString());
-
-                        // Only show the first 100 errors to prevent massive string allocations.
-                        if (i == 99)
-                        {
-                            Error.WriteLine($"And {errorCount - i + 1} more errors.");
-                            break;
-                        }
+                        Error.WriteLine($"And {errorCount - i + 1} more errors.");
+                        break;
```

Comment:
Should we set success = false here?

Reply:
It's technically possible to have 99 warnings too so this text may need to be updated.

Reply:
> Should we set success = false here?

Why? We want success to be true until we see a diagnostic with `Error` severity.

Reply:
So if you have 100 warnings and then an error it will .... ?

Reply:
Oh derp. Good catch 🥇. Will fix it.